Portal/vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/ApplicationModule.java

```diff
@@ -72,17 +74,31 @@ public void onFailure(Throwable caught) {
 
             @Override
             public void onSuccess(List<String> result) {
-             reservedClasses=result;
+             reservedGateClasses=result;
             }
         };
         ApplicationService.Util.getInstance().getAppletGateLabClasses(callback);
       
-  
+        final AsyncCallback<List<String>> callbackTest = new AsyncCallback<List<String>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                Layout.getInstance().setWarningMessage("Unable to load applet gatelab classes:<br />" + caught.getMessage());
+            }
+
+            @Override
+            public void onSuccess(List<String> result) {
+             reservedTestClasses=result;
+            }
+        };
+        ApplicationService.Util.getInstance().getAppletGateLabTestClasses(callbackTest);
+          
     }
 
     @Override
     public void load() {
-
+        reservedClasses = new ArrayList<String>();
+        reservedClasses.addAll(reservedGateClasses);
```

Comment:
At this point there is no guarantee that reservedGateClasses was initialized: the call to getApletGateLab* done in the constructor could still be in progress. This is a bug, but it was already present before your modifications.